### ISSUE TITLE
A question about the training process

### ISSUE THREAD

**Issue by yangye19960922** [chickenbestlover/Online-Recurrent-Extreme-Learning-Machine#5]
Hi, I have successfully ran through your code, but I have a question I want to communicate with you.

As shown in codes below, why you choose to compare the gap between predictedinput [i+1] and targetInput [i+1]  instead of comparing the gap between predictedinput [i+1] and trueData [i+1] .

I think predictedinput [i+1]  is the prediction at time i+1 which is correspond to trueData [i+1]

![image](https://user-images.githubusercontent.com/36871131/112573097-8a973980-8e26-11eb-883b-3db60cdd905b.png)

**Comment by ericleonardo** [comment]
Please, look if this solves your question:

https://github.com/chickenbestlover/Online-Recurrent-Extreme-Learning-Machine/issues/3#issuecomment-473770958

> In the concept of online learning, we don't need to divide the dataset into test and train.
Given a pair of an input and a target, the model first predicts an output using its corresponding input and then the model is trained using its output and the corresponding target. (training does not affect prediction. This is because the prediction is performed before training for all samples.)
In other words, we can say that new training samples are obtained as time progresses.
